rfcs/0022-kviklet.md

```diff
@@ -0,0 +1,86 @@
+  - Gather quantitative data about realtime tracking problems, instead of just ad hoc reports when somebody notices them.
+  - Sign in isn't working for only one operator. Do we have their info in the database?
+- Run Analytics. Examples:
+  - [What are people putting into free-text comments, and does it suggest any features we should build for them?](https://app.asana.com/0/616151179860796/1206468619394322)
+  - How long before a trip are inspectors entering departure times? (This affects prediction quality, and has been a major effort of Glides that we've been iterating on for a while, but it's harder when it's hard to export data.)
+
+This is not just a problem for Glides. Any app with a database is likely to run into similar problems.
+
+This has been a problem for us for a long time, which we haven't found a good solution for yet.
+
+## Status Quo / Alternative
+
+The prod database can only be accessed by the application, so in order to run a query, the Glides team must deploy a code change to the server. This is pretty high overhead that must be paid even for a small one-off query that takes 30 seconds to plan and run locally.
+
+Currently, the Glides app has an admin UI with a way to run a few hand-written queries. Adding or changing these is not easy, because it requires updaing UI and API endpoints in addition to writing the query itself.
+ 
+When [I asked other teams](https://app.asana.com/0/1200506724882024/1206647196651141/f), the recommended approach was to build a system like T-Alerts, where they still have to deploy code with a hardcoded approved query, but have a better system set up to reduce the effort and overhead in writing new queries. Glides was planning on doing this approach until we learned of Kviklet.
+
+Other options were considered and rejected in the asana task linked above.
+
+## What is Kviklet
+
+[Kviklet](https://kviklet.dev/) is a tool for reviewing and running queries against the prod datasbase. It's free, open source (MIT license), and self hosted. It requires review before running a query, similar to a pull request, and leaves an audit trail.
+
+It can be configured to require different amounts of review when touching different databases (for example, requiring extra review before accessing databases with PII).
+
+It supports Keycloak SSO (though we'd need to do role management within Kviklet).
+
+We would only give Kviklet read access to our databases, not write or admin access, to reduce risk. Any write queries would continue to have to go through an application deploy, like today.
+
+## Benefits of Kviklet compared to alternative
+
+- Will be much faster and easier to run queries.
+- Is full of features that are meant for database access, such as audit logs, previewing an query plan, saving queries and results, granting only temporary access, etc.
+- Would scale well to be usable by all teams, instead of each team needing to build its own custom UI for running queries.
```

Comment:
By "scale", I mean we can use the same centralized solution for all teams, instead of doing a bunch of custom effort for each project. I don't mean performance, or other definitions of "scaling", and you're right, it is untested.